8617 bug: link list incorrect markup

skibinska opened the pull request:
Relates https://github.com/wellcometrust/corporate/issues/8617

### Context

List link markup was incorrect - it was displaying `<div>` instead of `<ul>`.

<img src="https://user-images.githubusercontent.com/10700103/119974084-2ae71700-bfac-11eb-96c6-51bed7a5bc2f.png" />

I have made a change based on assumption that `link_list` variant is wrapping the `<li>` elements.

I created a [page, where I listed the list variants](https://develop.wellcome-corporate-fe.org.uk/what-we-do/our-work/listing-options).

### This PR

- adds conditional to check if the `list_link` option was selected then use `<ul>` as a wrapper

Commit:
8617 bug: adds condition to show ul as a wrapper

Commit:
8617 bug: removes incorrect markup in InfoBox story

skibinska commented on src/components/InfoBox/InfoBox.stories.tsx:
```diff
@@ -40,10 +40,8 @@ const InfoBoxExample = () => {
-        <Contact name="John Smith" />
-        <Contact name="Bertie Basset" />
-      </Listing>
+      <Contact name="John Smith" />
```
I don't think it is possible to add a Contact component within Listing.

(I based it on [the page I created](https://develop.wellcome-corporate-fe.org.uk/what-we-do/our-work/listing-options) and on the teams page where there is a lot of contact us within InfoBox and non of it is wrapped within `<Listing>` component, e.g. [International cultural programmes team](https://wellcome.org/who-we-are/teams/international-cultural-programmes-team)

creido-welly replied:
It may have been the case once upon a time but looking at the CMS I don't think it is possible.

skibinska commented on src/components/Listing/Listing.tsx:
```diff
@@ -37,7 +37,7 @@ export const Listing = ({
   className,
   variant = 'link_list'
```
I was wondering if we need this to be default, but I guess when we create a custom page and want to add a list of links, it could be easier to just add `<Listing>` without any props for the correct markup to be created.

skibinska commented on src/components/Listing/Listing.tsx:
```diff
@@ -37,7 +37,7 @@ export const Listing = ({
   variant = 'link_list'
 }: ListingProps) => {
-  const Element = as;
+  const Element = variant === 'link_list' ? 'ul' : as;
```
I am making an assumption that we want to create an unordered list. I don't think DCPs have an option to choose the wrapping element and I didn't see `<ol>` as a wrapper for a link list anywhere in the codebase.

creido-welly replied:
I _think_ it's a safe call but it would be good to test by creating/testing each variant, just in case.

creido-welly reviewed (approved):
Approved pending testing against each variant to ensure markup is correct

skibinska commented:
Thanks @creido-welly! I added these changes to my QA links branch. I added a page where I listed all of the listing options and I can confirm that everything looks ok :) 

Link to qa branch with listed listing options: https://qa-8554-yellow-links.wellcome-corporate-fe.org.uk/what-we-do/our-work/listing-options.

We can compare it to the develop version, where markup is incorrect: https://develop.wellcome-corporate-fe.org.uk/what-we-do/our-work/listing-options

creido-welly reviewed (approved):
All good to merge
